### PR TITLE
Cleanup hipProp's

### PR DESCRIPTION
**Pull request by arghdos**

### CODE REVIEW

File: core/src/HIP/Kokkos_HIP_Instance.cpp [hunk 1]
```diff
@@ -216,13 +216,14 @@ void HIPInternal::initialize(int hip_device_id, hipStream_t stream) {
...
     // theoretically, we can get 40 WF's / CU, but only can sustain 32
+    // see
+    // https://github.com/ROCm-Developer-Tools/HIP/blob/a0b5dfd625d99af7e288629747b40dd057183173/vdi/hip_platform.cpp#L742
     m_maxBlocksPerSM = 32;
     // FIXME_HIP - Nick to implement this upstream
-    m_regsPerSM          = 262144 / 32;
-    m_shmemPerSM         = hipProp.maxSharedMemoryPerMultiProcessor;
-    m_maxShmemPerBlock   = hipProp.sharedMemPerBlock;
-    m_maxThreadsPerSM    = m_maxBlocksPerSM * HIPTraits::WarpSize;
-    m_maxThreadsPerBlock = hipProp.maxThreadsPerBlock;
```
Comment (dalg24) [inline core/src/HIP/Kokkos_HIP_Instance.cpp:225]: Did you mean to not initialize `m_maxThreadsPerBlock` any more?

Comment (dalg24) [reply core/src/HIP/Kokkos_HIP_Instance.cpp:225]: I assume this goes away with #3299

Comment (arghdos) [reply core/src/HIP/Kokkos_HIP_Instance.cpp:225]: Yes -- I did the splitting a little... sub-optimally, we could merge this into #3299.

Comment (masterleinad) [reply core/src/HIP/Kokkos_HIP_Instance.cpp:225]: I think that's fair.

Comment (arghdos) [reply core/src/HIP/Kokkos_HIP_Instance.cpp:225]: Ok, merged into #3299

**Pull request closed without merging**